app/api/chats/[chatId]/route.js

```diff
@@ -0,0 +1,60 @@
+import Chat from "@/app/models/chat";
+import Message from "@/app/models/message";
+import User from "@/app/models/userser";
+import { connectMongoDB } from "@/app/lib/mongodb";
```

Comment:
The code is clear and effectively retrieves chat data. Standardize the database connection function (connectMongoDB() vs. connectToDB()). Error handling works, but more descriptive messages could improve debugging.